api/client/client.go

```diff
@@ -87,6 +87,15 @@ func (c *Client) Do(ctx context.Context, method, path string, body io.Reader) ([
 		return nil, err
 	}
 
+	// For 5xx errors, return a generic message instead of parsing the response
+	if res.StatusCode >= 500 && res.StatusCode <= 599 {
+		return nil, &APIError{
+			HTTPStatusCode: res.StatusCode,
+			Message: "Something went wrong. Try again shortly.",
+		}
+	}
```

Comment:
We can do this but my worry is we will gobble up the actual data payload that contained the reason for the issue. 

The error could be logged in the `cli.log` file for further debugging in future and hence should contain all the necessary upstream error information.

Wdyt ?

Reply:
What we can do is try and represent 2 kinds of error messages in the APIError maybe, one a user consumable and another one which is actual error message. This can then be caller's responsibility to check and display error appropriately.